internal/configs/configurator.go

```diff
@@ -551,7 +551,9 @@ func (cnf *Configurator) addOrUpdateTransportServer(transportServerEx *Transport
 	name := getFileNameForTransportServer(transportServerEx.TransportServer)
 
 	listener := cnf.globalCfgParams.Listeners[transportServerEx.TransportServer.Spec.Listener.Name]
-	tsCfg := generateTransportServerConfig(transportServerEx, listener.Port, cnf.isPlus)
+	tsc := newTransportServerConfigurator(cnf.isPlus, cnf.staticCfgParams)
+
+	tsCfg := generateTransportServerConfig(transportServerEx, listener.Port, tsc)
```

Comment:
warnings need to be propagated back to Controller and reported there

Reply:
We confined Warnings to `generateTransportServerConfig()` as the supporting infrastructure isn't available. Warnings are handled at a higher level and the TransportServer status is updated. Waiting on https://github.com/nginxinc/kubernetes-ingress/pull/1415 for this.